app/controllers/CheckYourAnswersController.scala

```diff
@@ -151,81 +156,143 @@ class CheckYourAnswersController @Inject()(
+        logger.error(s"Data missing - country of consumption from country ${countryFromIndex.position}")
+        Some(routes.CountryOfConsumptionFromEuController.onPageLoad(CheckMode, period, countryFromIndex, Index(0)))
+      case DataMissingError(VatRatesFromEuPage(countryFromIndex, countryToIndex)) =>
+        logger.error(s"Data missing - vat rates for sales from country ${countryFromIndex.position} to country ${countryToIndex.position}")
+        Some(routes.VatRatesFromEuController.onPageLoad(CheckMode, period, countryFromIndex, countryToIndex))
+      case DataMissingError(EuSalesAtVatRateQuery(countryFromIndex, countryToIndex, vatRateIndex)) =>
+        logger.error(s"Data missing - net value of sales from country ${countryFromIndex.position} to country " +
+          s"${countryToIndex.position} at vat rate ${vatRateIndex.position} ")
+        Some(routes.NetValueOfSalesFromEuController.onPageLoad(CheckMode, period, countryFromIndex, countryToIndex, vatRateIndex))
+      case DataMissingError(VatOnSalesFromEuQuery(countryFromIndex, countryToIndex, vatRateIndex)) =>
+        logger.error(s"Data missing - vat charged on sales from country ${countryFromIndex.position} to country " +
+          s"${countryToIndex.position} at vat rate ${vatRateIndex.position} ")
+        Some(routes.VatOnSalesFromEuController.onPageLoad(CheckMode, period, countryFromIndex, countryToIndex, vatRateIndex))
 
+      case DataMissingError(AllCorrectionPeriodsQuery) =>
+        logger.error(s"Data missing - no data provided for corrections")
+        Some(correctionsRoutes.CorrectionReturnPeriodController.onPageLoad(CheckMode, period, Index(0)))
+      case DataMissingError(AllCorrectionCountriesQuery(periodIndex)) =>
+        logger.error(s"Data missing - no countries found for corrections to period ${periodIndex.position}")
+        Some(correctionsRoutes.CorrectionCountryController.onPageLoad(CheckMode, period, periodIndex, Index(0)))
+      case DataMissingError(CorrectionToCountryQuery(periodIndex, countryIndex)) =>
+        logger.error(s"Data missing - correction to country ${countryIndex.position} in period ${periodIndex.position}")
+        Some(correctionsRoutes.CountryVatCorrectionController.onPageLoad(CheckMode, period, periodIndex, countryIndex))
+
+      case DataMissingError(_) =>
+        logger.error(s"Unhandled DataMissingError")
+        None
+      case _ =>
+        logger.error(s"Unhandled ValidationError")
+        None
+    }
+  }
+
+  def submitReturn(vatReturnRequest: VatReturnRequest, correctionRequest: Option[CorrectionRequest], period: Period)
+                  (implicit request: DataRequest[AnyContent]): Future[Result] = {
+
+    val submission = correctionRequest match {
+      case Some(cr) => vatReturnConnector.submit(VatReturnWithCorrectionRequest(vatReturnRequest, cr))
+      case _ => vatReturnConnector.submit(vatReturnRequest)
+    }
+
+    submission.flatMap {
```

Comment:
The first 2 cases have the same outcome, could refactor the order to make it so we just have one? Also the tabbing is a bit out on this, makes it a little harder to read